### PR TITLE
[warnings] fix lots of bugprone warnings

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/emu/scard/smartcard_emulate.c [hunk 1]
```diff
@@ -1253,7 +1253,7 @@ HANDLE WINAPI Emulate_SCardAccessStartedEvent(SmartcardEmulationContext* smartca
 	WLog_Print(smartcard->log, smartcard->log_default_level, "SCardAccessStartedEvent {");
 
 	/* Not required, return random */
-	winpr_RAND(&hEvent, sizeof(hEvent));
+	winpr_RAND((void*)&hEvent, sizeof(hEvent));
```
Comment (hardening) [inline libfreerdp/emu/scard/smartcard_emulate.c:1257]: Strange I would have bet that cast to `void *` was implicit in C.

**Review by hardening — COMMENTED** [review]
I don't see the added value to the adding of `(void*)` casts, except that it tends to bloat the code. And once this part of the patch will have passed we will have to use that everywhere at every new PR...

File: libfreerdp/emu/scard/smartcard_emulate.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/emu/scard/smartcard_emulate.c:1257]: No, the only place these casts added are `void**` to `void*` or similar casts.

While implicit like integer narrow/widen it is (as the title suggests) often a source for buggy code (pass wrong type to function) and for that reason not necessarily without warnings

File: cmake/CleaningConfigureFile.cmake [hunk 2]
```diff
@@ -6,6 +6,6 @@
 
 function(cleaning_configure_file SRC DST)
   configure_file(${SRC} ${DST} ${ARGN})
-  set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_CLEAN_FILES ${DST})
+  # set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_CLEAN_FILES ${DST})
```
Comment (mfleisz) [inline cmake/CleaningConfigureFile.cmake:9]: Intended or debugging leftover?

Comment (mfleisz) [inline cmake/CleaningConfigureFile.cmake:9]: Intended or debugging leftover?

**Review by mfleisz — COMMENTED** [review]
Overall LGTM - just (maybe?) one leftover in the cmake config file

File: cmake/CleaningConfigureFile.cmake (hunk 2 above)
Comment (akallabeth) [reply cmake/CleaningConfigureFile.cmake:9]: true, that should be removed.

Comment (akallabeth) [reply cmake/CleaningConfigureFile.cmake]: and fixed.

**Comment by akallabeth** [comment]
@hardening the `(void*)` casts are there for the specific parts of code where we do some 2d array magic. (allocate with one `malloc` and `free`, ...)
while this works fine it is in no way the intended usage of the `API` and therefore prone to emit warnings.

**Pull request merged**